docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+
+``create_pep_zero`` hook is called. See `4. PEP 0 generation`_.
+
+
+3.4 Read document
+~~~~~~~~~~~~~~~~~
+
+Parsing the document is handled by ``PEPParser``
+(``pep_sphinx_extensions.pep_processor.parsing.pep_parser.PEPParser``), a light
+wrapper over ``sphinx.parsers.RSTParser``.
+
+``PEPParser`` serves two purposes, firstly to mark the document as containing
+:rfc:`2822` headers, and secondly to register the transforms we want to apply.
+These are:
+
+- ``PEPHeaders``
+- ``PEPTitle``
+- ``PEPContents``
+- ``PEPFooter``
+
+Transforms are then applied in priority order.
+
+
+3.4.1 ``PEPRole`` role
+**********************
+
+This overrides the built-in ``:pep:`` role to return the correct URL.
+
+
+3.4.2 ``PEPHeaders`` transform
+******************************
+
+PEPs start with a set of :rfc:`2822` headers, per :doc:`PEP 1 <../pep-0001>`.
+This transform validates that the required headers are present and of the
+correct data type, and removes headers not for display.
+It must run before the ``PEPTitle`` transform.
+
+
+3.4.3 ``PEPTitle`` transform
+****************************
+
+We generate the title node from the parsed title in the PEP headers, and make
+all nodes in the document children of the new title node.
+This transform must also handle parsing reStructuredText markup within PEP
+titles, such as :doc:`PEP 604 <../pep-0604>`.
+
+
+3.4.4 ``PEPContents`` transform
+*******************************
+
+The automatic table of contents (TOC) is inserted in this transform in a two
+part process.
+
+Firstly the transform inserts a placeholder for the TOC and a horizontal rule
+after the document title and PEP headers.
+Secondly, a callback transform recursively walks the document to create the TOC,
+starting from after the placeholder node.
+Whilst walking the document all reference nodes in the titles are removed, and
+titles are given a self-link.
+
+
+3.4.5 ``PEPFooter`` transform
+*****************************
+
+This firstly builds a map of file modification times from a single git call, as
+a speed-up. This will return incorrect results on a shallow check-out of the
+repository, as is the default on continuous integration systems.
+
+We then attempt to remove any empty references sections, and append metadata in
+the footer (source link and last modified timestamp).
+
+
+3.5 Prepare for writing
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+``pep_html_builder.FileBuilder.prepare_writing`` initialises the bare miniumum
+of the Docutils writer and the settings for writing documents.
+This provides a significant speed-up over the base Sphinx implementation as most
+of the data automatically initialised was unused.
+
+
+3.6 Translate Docutils to HTML
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``PEPTranslator`` overrides paragraph and reference logic to replicate
+processing from the previous ``docutils.writers.pep`` based system.
+Paragraphs are to be made compact where possible by omitting <p> tags, and
+footnote references should be enclosed in square brackets.
```

Comment:
```suggestion
Paragraphs are made compact where possible by omitting ``<p>`` tags, and
footnote references are be enclosed in square brackets.
```

Simplify and correct grammar/phrasing, and use verbatim on `<p>`; this describes what the translator actually does, not just what it is planned to do.